### PR TITLE
dYdX Rewards adapter

### PR DESCRIPTION
**Pull request by boxhock**
......

## Changes

- Adds IPFS adapter
- Adds dYdX rewards adapter
- Modified `Dockerfile` to support IPFS

## Steps to Test

1. Integration tests and e2e tests included where appropriate

## Quality Assurance

- [x] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `<ADAPTER_PACKAGE>/schemas/env.json` and `<ADAPTER_PACKAGE>/README.md`
- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `infra-k8s` configuration file.
- [x] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Clubhouse/Shortcut
- [x] This is related to a maximum of one Clubhouse/Shortcut story or GitHub issue
- [ ] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types)

### CODE REVIEW

**Review by justinkaseman — DISMISSED** [review]
Some really high quality work here. Nice stuff! Tests are great.

Before merging - is this something we want to be deployed on staging for soak testing?

File: packages/sources/ipfs/src/endpoint/read.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  if (!cid) {
+    for await (const ipfsCid of client.name.resolve(`/ipns/${ipns}`)) {
+      cid = ipfsCid.replace('/ipfs/', '')
+    }
+  }
+
+  const codec = validator.validated.data.codec
+
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  let result: string | object
```
Comment (justinkaseman) [inline packages/sources/ipfs/src/endpoint/read.ts:45]: Could we use something like 
```
{ [key: string]: unknown }
```
on these unknown object types?

@HenryNguyen5 do you know of a better way TypeScript master?

**Comment by boxhock** [comment]
@justinkaseman We're not going to deploy this in our infra, and soak testing is a bit difficult with this adapter as each request is supposed to make an on-chain transaction. The goal was to have good enough integration tests with mocks to make up for that :sweat_smile:

File: packages/sources/ipfs/src/endpoint/read.ts (hunk 1 above)
Comment (HenryNguyen5) [reply packages/sources/ipfs/src/endpoint/read.ts:45]: Looks like the `object` type is being propagated from the ipfs library, to make it more apparent i've suggested changed via using referenced types in https://github.com/smartcontractkit/external-adapters-js/pull/921

**Review by justinkaseman — APPROVED** [review]

**Pull request merged**